library/gen_x12_837.inc.php

```diff
@@ -38,236 +43,152 @@ function gen_x12_837($pid, $encounter, &$log, $encounter_claim = false)
     "*" . $claim->billingContactName() .
-    "*TE" .
-    "*" . $claim->billingContactPhone();
-    if (!$CMS_5010 && $claim->x12gsper06()) {
-        $out .= "*ED*" . $claim->x12gsper06();
-    }
-
+    "*" . "TE" .
+    "*" . $claim->billingContactPhone() .
+    "*" . "EM" .
+    "*" . $claim->billingContactEmail();
     $out .= "~\n";
 
     ++$edicount;
-    $out .= "NM1" .       // Loop 1000B Receiver
-    "*40" .
-    "*2" .
+    $out .= "NM1" .    // Loop 1000B Receiver
+    "*" . "40" .
+    "*" . "2" .
     "*" . $claim->clearingHouseName() .
     "*" .
     "*" .
     "*" .
     "*" .
-    "*46" .
+    "*" . "46" .
     "*" . $claim->clearingHouseETIN() .
     "~\n";
 
-    $HLcount = 1;
-
+    ++$HLcount;
     ++$edicount;
     $out .= "HL" .        // Loop 2000A Billing/Pay-To Provider HL Loop
-    "*$HLcount" .
+    "*" . $HLcount .
     "*" .
-    "*20" .
-    "*1" .              // 1 indicates there are child segments
+    "*" . "20" .
+    "*" . "1" .              // 1 indicates there are child segments
     "~\n";
 
     $HLBillingPayToProvider = $HLcount++;
-
-  // Situational PRV segment for provider taxonomy code for Medicaid.
+    // Situational PRV segment for provider taxonomy code for Medicaid.
     if ($claim->claimType() == 'MC') {
         ++$edicount;
-        $out .= "PRV*BI*ZZ" .
+        $out .= "PRV" .
+        "*" . "BI" .
+        "*" . "ZZ" .
         "*" . $claim->providerTaxonomy() .
         "~\n";
     }
 
-  // Situational CUR segment (foreign currency information) omitted here.
-
+    // Situational CUR segment (foreign currency information) omitted here.
     ++$edicount;
-  //Field length is limited to 35. See nucc dataset page 63 www.nucc.org
     if ($claim->federalIdType() == "SY") { // check for entity type like in 1000A
-        $tempName = $claim->billingFacilityName();
-        $partsName = explode(' ', $tempName);// Loop 2010AA Billing Provider entity == person
-        $num_parts = count($partsName);
-        switch ($num_parts) {
-            case "2":
-                $firstName = $partsName[0];
-                $middleName = '';
-                $lastName = $partsName[1];
-                $suffixName = '';
-                break;
-            case "3":
-                $firstName = $partsName[0];
-                $middleName = $partsName[1];
-                $lastName = $partsName[2];
-                $suffixName = '';
-                break;
-            case "4":
-                $firstName = $partsName[0];
-                $middleName = $partsName[1];
-                $lastName = $partsName[2];
-                $suffixName = $partsName[3];
-                break;
-            default:
-                $log .= "*** billing provider name in 2010AA loop has more than 4 parts, may not be desirable\n";
-                $firstName = $partsName[0];
-                $middleName = $partsName[1];
-                $lastName = $partsName[2];
-                $suffixName = $partsName[3];
```

Comment:
$suffixName now looks like a loose end below. Should we remove this or support it by collecting the provider suffix?